File: module-simple/src/main/java/domainapp/modules/simple/fixture/SimpleObjectBuilder.java
package domainapp.modules.simple.fixture;

import javax.inject.Inject;

import org.apache.isis.testing.fixtures.applib.fixturescripts.BuilderScriptWithResult;

import org.apache.isis.domox.modules.ksimple.dom.so.SimpleObject;
import org.apache.isis.domox.modules.ksimple.dom.so.SimpleObjects;
import lombok.Getter;
import lombok.Setter;
import lombok.experimental.Accessors;

@Accessors(chain = true)
public class SimpleObjectBuilder extends BuilderScriptWithResult<SimpleObject> {

    @Getter @Setter
    private String name;

    @Override
    protected SimpleObject buildResult(final ExecutionContext ec) {

        checkParam("name", ec, String.class);

        return wrap(simpleObjects).create(name);
    }

    // -- DEPENDENCIES

    @Inject SimpleObjects simpleObjects;

}

File: module-ksimple/src/main/kotlin/org.apache.isis.domox/modules/ksimple/fixture/SimpleObjectBuilder.kt
package  org.apache.isis.domox.modules.ksimple.fixture

import org.apache.isis.domox.modules.ksimple.dom.so.SimpleObject
import org.apache.isis.domox.modules.ksimple.dom.so.SimpleObjects
import org.apache.isis.testing.fixtures.applib.fixturescripts.BuilderScriptWithResult
import javax.inject.Inject

class SimpleObjectBuilder : BuilderScriptWithResult<SimpleObject?>() {
    private var name: String? = null
    fun setName(value: String): SimpleObjectBuilder {
        name = value
        return this
    }

    override fun buildResult(ec: ExecutionContext?): SimpleObject {
        checkParam("name", ec, String::class.java)
        return wrap(simpleObjects)!!.create(name)
    }

    // -- DEPENDENCIES
    @Inject
    var simpleObjects: SimpleObjects? = null

}
